modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/dml/DmlUtils.java

```diff
@@ -371,10 +370,7 @@ private static UpdateResult doDelete(GridCacheContext cctx, Iterable<List<?>> cu
 
             Object key = row.get(0);
 
-            ClusterNode node = snd.primaryNodeByKey(key);
-
-            IgniteInClosure<MutableEntry<Object, Object>> rmvC =
-                DmlStatementsProcessor.getRemoveClosure(node, key);
+            IgniteInClosure<MutableEntry<Object, Object>> rmvC = new DmlStatementsProcessor.RemoveClosure();
```

Comment:
Let's continue to reuse one instance of remove closure like we do now.

Reply:
> Let's continue to reuse one instance of remove closure like we do now.

Thanks, fixed.